assignments/closure.js

```diff
@@ -1,21 +1,147 @@
+logHeader('CHALLENGE 1: Write your own closure', 'Write a simple closure of your own creation. Keep it simple! (nahh..)');
 
-// ==== Challenge 2: Create a counter function ====
+/* 
+  Write a simple closure of your own creation.  Keep it simple!
+*/
+
+console.log('See consoleCannon.js & ../consoleCannon/');
+
+
+
+/*********************************************************************************
+*                        CHALLENGE 2: Create a counter function                  *
+**********************************************************************************/ 
+logHeader('STRETCH CHALLENGE 2: Create a counter function');
 const counter = () => {
-  // Return a function that when invoked increments and returns a counter variable.
+  let count = 0;
+  
+  // Return a function that when invoked increments and returns a counter variable
+  return function () {
+    return count++;
+  }
 };
-// Example usage: const newCounter = counter();
-// newCounter(); // 1
-// newCounter(); // 2
 
-// ==== Challenge 3: Create a counter function with an object that can increment and decrement ====
+const newCounter = counter();
+
+console.log(newCounter()); // 1
+console.log(newCounter()); // 2
+console.log(newCounter()); // 3
+console.log(newCounter()); // 4
+console.log(newCounter()); // 5
+
+
+
+/*********************************************************************************
+* CHALLENGE 3: Create a counter function w/object that increments and decrements *
+**********************************************************************************/ 
+logHeader('CHALLENGE 3: Create a counter function w/object that increments and decrements', 'Return an object that has two methods called \'increment\' and \'decrement\'.\n\'increment\' should increment a counter variable in closure scope and return it.\n\'decrement\' should decrement the counter variable and return it.');
+
+/*
+  Return an object that has two methods called `increment` and `decrement`.
+  `increment` should increment a counter variable in closure scope and return it.
+  `decrement` should decrement the counter variable and return it.
+*/
+
 const counterFactory = () => {
-  // Return an object that has two methods called `increment` and `decrement`.
-  // `increment` should increment a counter variable in closure scope and return it.
-  // `decrement` should decrement the counter variable and return it.
+  let count = 0;
+
+  return {
+    increment: function() {
+      return count++;
+    },
+
+    decrement: function () {
+      return count--;
+    }
+  }
 };
+
+
+// counterFactory is invoked inside randomCounterFactory below
+const randomCounterFactory = () => {
+
+  const counter = counterFactory();
+
+  const ticker = {
+    
+    // string representation of the 'ticker'
+    string: `                                       .`,
+
+    // removes first space of the ticker string
+    shiftTicker: function () {
```

Comment:
The way you visualized your `randomCounterFactory` is so good. I love the tug of war in my browser console.